libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/targets/native/spm/SwiftPackageExtension.kt

```diff
@@ -0,0 +1,380 @@
+        }
+
+        fun testTarget(name: String, configure: Target.() -> Unit) {
+            val target = Target(name).apply(configure)
+            tests.add(target)
+        }
+
+        fun systemLibrary(name: String, configure: SystemLibrary.() -> Unit) {
+            val library = SystemLibrary(name).apply(configure)
+            libraries.add(library)
+        }
+
+        fun binaryTarget(name: String, url: String, checksum: String) {
+            val binaryTarget = BinaryTarget(name, url, checksum)
+            binaries.add(binaryTarget)
+        }
+
+        fun binaryTarget(name: String, path: String) {
+            val binaryTarget = BinaryTarget(name, path)
+            binaries.add(binaryTarget)
+        }
+
+        /** TODO
+         * Все target оказались одинаковыми по сигнатурам,
+         * кажется нет смысла разносить по разным классам
+         */
+        class Target(@Input val name: String) {
+            @Input
+            @Optional
+            var path: String? = null
+
+            @Input
+            @Optional
+            var publicHeadersPath: String? = null
+
+            @Nested
+            var dependenciesTarget = mutableListOf<TargetDependencies.Target>()
+
+            @Nested
+            var dependenciesProduct = mutableListOf<TargetDependencies.Product>()
+
+            @Nested
+            var excludePaths = mutableListOf<String>()
+
+            @Nested
+            @Optional
+            var sourceFiles: List<String>? = null
+
+            @Nested
+            @Optional
+            var resources: List<Resource>? = null
+
+            fun dependencies(configure: TargetDependencies.() -> Unit) {
+                val dependency = TargetDependencies().apply(configure)
+                dependenciesTarget = dependency.targets
+                dependenciesProduct = dependency.products
+            }
+
+            fun exclude(configure: ExcludePath.() -> Unit) {
+                val exclude = ExcludePath().apply(configure)
+                excludePaths = exclude.paths
+            }
+
+            fun sources(configure: SourceFile.() -> Unit) {
+                val sources = SourceFile().apply(configure)
+                sourceFiles = sources.files
+            }
+
+            fun resources(configure: Resource.() -> Unit) {
+                val resource = Resource().apply(configure)
+                resources = resource.resources
+            }
+
+            // TODO: доделать
+//            fun cSettings() {}
+//            fun cxxSettings() {}
+//            fun swiftSettings() {}
+//            fun linkerSettings() {}
+
+            class Resource() {
+                var resources = mutableListOf<Resource>()
+
+                @Input
+                lateinit var path: String
+
+                // TODO: непонятно как задается
+                @Input
+                @Optional
+                var localization: Localization? = null
+
+                constructor(path: String) : this() {
```

Comment:
Try to replace multiple constructors with default values or introduce builder for creating instances.